tests/high-level-tests.lisp

```diff
@@ -26,10 +26,30 @@
     (is (magicl:unitaryp (magicl:random-unitary i)))))
 
 (deftest test-logm ()
-  "Check that the matrix logarithm is the inverse of the matrix exponential"
+  "Check that the matrix logarithm is the inverse of the matrix exponential."
   (let ((x (magicl:random-unitary 4)))
     (loop :for i :from 0 :to (1- (magicl:matrix-cols x))
 	  :for j :from 0 :to (1- (magicl:matrix-rows x))
 	  :do (let  ((diff (- (magicl:ref x i j) (magicl:ref (magicl:expm (magicl:logm x)) i j)))
                       (eps .00001f0))
                 (is (< (abs diff) eps))))))
+
+(deftest test-kron ()
+  "Test a few properties of the kronecker product."
+  (let* ((matrix-dim 2)
+         (eye2 (make-identity-matrix matrix-dim))
+         (eye4 (kron eye2 eye2))
+         (x (magicl:make-complex-matrix matrix-dim matrix-dim '(0 1 1 0)))
+         (xxx (kron x x x))
+         (xxx-cols (matrix-cols xxx)))
+    ;; Check that the kronecker product of two small identities is a larger identity.
+    (is (magicl:identityp eye4))
+    ;; Check that x\otimes xotimes x is evaluated correctly.
+    (is (loop :for i :below (matrix-rows xxx) :always
+                                              (loop :for j :below xxx-cols
+                                                    :always (if (= (- xxx-cols 1) (+ i j))
+                                                                (= 1 (ref xxx i j))
+                                                                (= 0 (ref xxx i j))))))
+    ;; Check that I\otimes X != X\otimes I
+    (is (not (equalp (kron eye2 x) (kron x eye2))))
```

Comment:
This is a step in the right direction but I would like you to also compare one of the two expressions against the "analytical result".

Reply:
Fair enough!